dispatches/case_studies/renewables_case/run_double_loop_battery.py

```diff
@@ -0,0 +1,308 @@
+    )
+
+    parser.add_argument(
+        "--n_scenario",
+        dest="n_scenario",
+        help="Set the number of price scenarios.",
+        action="store",
+        type=int,
+        default=3,
+    )
+
+    parser.add_argument(
+        "--participation_mode",
+        dest="participation_mode",
+        help="Indicate the market participation mode.",
+        action="store",
+        type=str,
+        default="Bid",
+    )
+
+    options = parser.parse_args()
+
+    sim_id = options.sim_id
+    wind_pmax = options.wind_pmax
+    battery_energy_capacity = options.battery_energy_capacity
+    battery_pmax = options.battery_pmax
+    n_scenario = options.n_scenario
+    participation_mode = options.participation_mode
+    p_min = 0
+
+    allowed_participation_modes = {"Bid", "SelfSchedule"}
+    if participation_mode not in allowed_participation_modes:
+        raise ValueError(
+            f"The provided participation mode {participation_mode} is not supported."
+        )
+
+    wind_df = read_rts_gmlc_wind_inputs(rts_gmlc.source_data_path, wind_generator)
+    wind_df = wind_df[wind_df.index >= start_date]
+    gen_capacity_factor = wind_df[f"{wind_generator}-RTCF"].values.tolist()
+
+    # NOTE: `rts_gmlc_data_dir` should point to a directory containing RTS-GMLC scenarios
+    rts_gmlc_data_dir = rts_gmlc.source_data_path
+    output_dir = Path(f"new_wind_battery_ratio_duration_sweep_sb/battery_duration_{duration}/Sweep_rerun_wind_battery_stochastic_bidder_sim_{sim_id}")
+    # output_dir = Path(sim_name)
+
+    solver = pyo.SolverFactory(solver_name)
+
+    if participation_mode == "Bid":
+        thermal_generator_params = {
+            "gen_name": wind_generator,
+            "bus": bus_name,
+            "p_min": p_min,
+            "p_max": wind_pmax,
+            "min_down_time": 0,
+            "min_up_time": 0,
+            "ramp_up_60min": 1000,
+            "ramp_down_60min": 1000,
+            "shutdown_capacity": 1000,
+            "startup_capacity": 1000,
+            "initial_status": 1,
+            "initial_p_output": 0,
+            "production_cost_bid_pairs": [(p_min, 0), (wind_pmax + battery_pmax, 0)],
+            "include_default_p_cost": False,
+            "startup_cost_pairs": [(0, 0)],
+            "fixed_commitment": None,
+            "non_spinning_capacity": 0,
+            "supplemental_spinning_capacity": 0,
+            "supplemental_non_spinning_capacity": 0,
+            "agc_capable": False
+        }
+        model_data = ThermalGeneratorModelData(**thermal_generator_params)
+    elif participation_mode == "SelfSchedule":
+        generator_params = {
+            "gen_name": wind_generator,
+            "bus": bus_name,
+            "p_min": p_min,
+            "p_max": wind_pmax,
+            "p_cost": 0,
+            "fixed_commitment": None,
+        }
+        model_data = RenewableGeneratorModelData(**generator_params)
+
+    historical_da_prices = {
+        bus_name: [
```

Comment:
Where did these data come from? Are these data just place holders or do the results change if the data are changed? Please elaborate in a comment in the code.